File: leetcode/js/minimum-average-of-smallest-and-largest-elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumAverage = function (nums) {
  nums.sort((a, b) => a - b);

  let start = 0;
  let end = nums.length - 1;
  const averages = [];

  while (start < end) {
    const sum = nums[start] + nums[end];
    averages.push(sum / 2);
    start++;
    end--;
  }

  return Math.min(...averages);
};

File: leetcode/python/minimum-average-of-smallest-and-largest-elements.py
from collections import deque

class Solution:
    def minimumAverage(self, nums: list[int]) -> float:
        result = 51
        nums.sort()
        nums_q = deque(nums)

        while nums_q:
            min_el = nums_q.popleft()
            max_el = nums_q.pop()

            result = min(result, (min_el + max_el) / 2)

        return result